moveit_ros/manipulation/pick_place/src/place.cpp

```diff
@@ -51,6 +51,20 @@ PlacePlan::PlacePlan(const PickPlaceConstPtr& pick_place) : PickPlacePlanBase(pi
 
 namespace
 {
+struct OrderPlaceLocationQuality
+{
+  OrderPlaceLocationQuality(const std::vector<moveit_msgs::PlaceLocation>& places) : places_(places)
+  {
+  }
+
+  bool operator()(const std::size_t a, const std::size_t b) const
+  {
+    return places_[a].quality > places_[b].quality;
```

Comment:
To preserve order of equal-quality elements:
```suggestion
    if (places_[a].quality > places_[b].quality) return true;
    if (places_[a].quality == places_[b].quality) return a > b;
    return false;
```